tests/ut/language/parser/test_system_ops.py

```
@@ -189,6 +189,31 @@ def test_aiv_initialize_pipe_round_trip(self):
+    def test_initialize_pipe_rejects_non_positive_slot_num(self):
+        """A non-positive slot_num is rejected at the DSL boundary."""
+        with pytest.raises(ValueError, match="slot_num must be positive"):
+            pl.aic_initialize_pipe(dir_mask=1, slot_size=256, slot_num=0)
+        with pytest.raises(ValueError, match="slot_num must be positive"):
+            pl.aiv_initialize_pipe(dir_mask=2, slot_size=512, slot_num=-4)
+
+    def test_initialize_pipe_rejects_invalid_local_slot_num(self):
+        """local_slot_num must be positive and not exceed slot_num."""
+        with pytest.raises(ValueError, match="local_slot_num must be positive"):
+            pl.aic_initialize_pipe(dir_mask=1, slot_size=256, local_slot_num=0)
+        with pytest.raises(ValueError, match="local_slot_num .* must be <= slot_num"):
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)